### DISCUSSION TITLE
posts/setup-nextjs-with-airbnb-eslint-prettier-typescript-and-tailwindcss/

### DISCUSSION THREAD

**Discussion by giscus** [m4xshen/m4xshen.dev#6] [Announcements]
# posts/setup-nextjs-with-airbnb-eslint-prettier-typescript-and-tailwindcss/

Set up Next.js with Airbnb ESLint, Prettier, TypeScript, and Tailwind CSS for efficient development. A step-by-step guide for a seamless environment.

https://m4xshen.dev/posts/setup-nextjs-with-airbnb-eslint-prettier-typescript-and-tailwindcss/

**Comment by henrybabbage** [comment]
I got this error at the end with the above config:

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=4.3.5 <5.4.0

YOUR TYPESCRIPT VERSION: 5.4.5

**Reply by m4xshen** [reply]
@henrybabbage Seems that `@typescript-eslint` has newer version. You can try to update it with following command:

If your are using `npm`:
```bash
npm install eslint-config-airbnb-typescript \
            @typescript-eslint/eslint-plugin@^7.0.0 \
            @typescript-eslint/parser@^7.0.0 \
            --save-dev
```

If your are using `yarn`:
```bash
yarn add eslint-config-airbnb-typescript \
            @typescript-eslint/eslint-plugin@^7.0.0 \
            @typescript-eslint/parser@^7.0.0 \
            --dev
```

Please tell me whether this works. I'll then update the article!
